django_app/redbox_app/redbox_core/models.py

```diff
@@ -55,22 +54,39 @@ class ProcessingStatusEnum(models.TextChoices):
 
 
 class File(UUIDPrimaryKeyBase, TimeStampedModel):
-    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, null=False, blank=False)
-    original_file = models.FileField(storage=settings.BUCKET_NAME)
+    processing_status = models.CharField(
+        choices=ProcessingStatusEnum.choices, null=False, blank=False
+    )
+    original_file = models.FileField(storage=settings.DEFAULT_FILE_STORAGE)
     user = models.ForeignKey(User, on_delete=models.CASCADE)
+    original_file_name = models.TextField(max_length=2048, blank=True, null=True)
 
     @property
-    def file_type(self):
+    def file_type(self) -> str:
         name = self.original_file.name
         return name.split(".")[-1]
 
     @property
-    def file_url(self):
+    def url(self) -> str:
```

Comment:
Could we use a furl.URL here?

Reply:
Changed this to use furl now, I've checked the template still works fine